### DISCUSSION TITLE
How to build with gcc 4.8.5

### DISCUSSION THREAD

**Discussion by Joezczhang** [rdkit/rdkit#4459] [Q&A]
Hello:

when I build rdkit with gcc version 4.8.5,there are so many error just as below.how to solve it ?

/home/postgres/pgsource/external/rdkit/Code/GraphMol/CIPLabeler/CIPMol.cpp: In constructor 'RDKit::CIPLabeler::CIPMol::CIPMol(RDKit::ROMol&':
/home/postgres/pgsource/external/rdkit/Code/GraphMol/CIPLabeler/CIPMol.cpp:19:39: error: invalid initialization of non-const reference of type 'RDKit::ROMol&' from an rvalue of type '<brace-enclosed initializer list>'
 CIPMol::CIPMol(ROMol &mol) : d_mol{mol} {}
                                       ^
In file included from /home/postgres/pgsource/external/rdkit/Code/GraphMol/CIPLabeler/CIPMol.cpp:14:0:
/home/postgres/pgsource/external/rdkit/Code/GraphMol/CIPLabeler/CIPMol.h: In instantiation of 'RDKit::CIPLabeler::CIPMolSpan<T, U>::CIPMolSpan(RDKit::ROMol&, std::pair<OutEdgeIterator, OutEdgeIterator>&&) [with T = RDKit::Bond*; U = boost::detail::out_edge_iter<__gnu_cxx::__normal_iterator<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*>*, std::vector<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*>, std::allocator<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*> > > >, long unsigned int, boost::detail::edge_desc_impl<boost::undirected_tag, long unsigned int>, long int>]':
/home/postgres/pgsource/external/rdkit/Code/GraphMol/CIPLabeler/CIPMol.cpp:42:42:   required from here
/home/postgres/pgsource/external/rdkit/Code/GraphMol/CIPLabeler/CIPMol.h:53:37: error: invalid initialization of non-const reference of type 'RDKit::ROMol&' from an rvalue of type '<brace-enclosed initializer list>'
         d_iend{std::move(itr.second)} {}
                                     ^
/home/postgres/pgsource/external/rdkit/Code/GraphMol/CIPLabeler/CIPMol.h: In instantiation of 'RDKit::CIPLabeler::CIPMolSpan<T, U>::CIPMolSpan(RDKit::ROMol&, std::pair<OutEdgeIterator, OutEdgeIterator>&&) [with T = RDKit::Atom*; U = boost::adjacency_iterator<boost::adjacency_list<boost::vecS, boost::vecS, boost::undirectedS, RDKit::Atom*, RDKit::Bond*>, long unsigned int, boost::detail::out_edge_iter<__gnu_cxx::__normal_iterator<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*>*, std::vector<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*>, std::allocator<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*> > > >, long unsigned int, boost::detail::edge_desc_impl<boost::undirected_tag, long unsigned int>, long int>, long int>]':
/home/postgres/pgsource/external/rdkit/Code/GraphMol/CIPLabeler/CIPMol.cpp:47:46:   required from here
/home/postgres/pgsource/external/rdkit/Code/GraphMol/CIPLabeler/CIPMol.h:53:37: error: invalid initialization of non-const reference of type 'RDKit::ROMol&' from an rvalue of type '<brace-enclosed initializer list>'
In file included from /home/postgres/pgsource/external/rdkit/Code/GraphMol/CIPLabeler/Mancude.cpp:24:0:
/home/postgres/pgsource/external/rdkit/Code/GraphMol/CIPLabeler/CIPMol.h: In instantiation of 'RDKit::CIPLabeler::CIPMolSpan<T, U>::CIPMolIter::CIPMolIter(RDKit::ROMol&, U) [with T = RDKit::Bond*; U = boost::detail::out_edge_iter<__gnu_cxx::__normal_iterator<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*>*, std::vector<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*>, std::allocator<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*> > > >, long unsigned int, boost::detail::edge_desc_impl<boost::undirected_tag, long unsigned int>, long int>]':
/home/postgres/pgsource/external/rdkit/Code/GraphMol/CIPLabeler/CIPMol.h:55:47:   required from 'RDKit::CIPLabeler::CIPMolSpan<T, U>::CIPMolIter RDKit::CIPLabeler::CIPMolSpan<T, U>::begin() [with T = RDKit::Bond*; U = boost::detail::out_edge_iter<__gnu_cxx::__normal_iterator<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*>*, std::vector<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*>, std::allocator<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*> > > >, long unsigned int, boost::detail::edge_desc_impl<boost::undirected_tag, long unsigned int>, long int>]'
/home/postgres/pgsource/external/rdkit/Code/GraphMol/CIPLabeler/Mancude.cpp:42:46:   required from here
/home/postgres/pgsource/external/rdkit/Code/GraphMol/CIPLabeler/CIPMol.h:29:69: error: invalid initialization of non-const reference of type 'RDKit::ROMol&' from an rvalue of type '<brace-enclosed initializer list>'
     CIPMolIter(ROMol &mol, U pos) : d_mol{mol}, d_pos{std::move(pos)} {}
                                                                     ^
/home/postgres/pgsource/external/rdkit/Code/GraphMol/CIPLabeler/CIPMol.h: In instantiation of 'RDKit::CIPLabeler::CIPMolSpan<T, U>::CIPMolIter::CIPMolIter(RDKit::ROMol&, U) [with T = RDKit::Atom*; U = boost::adjacency_iterator<boost::adjacency_list<boost::vecS, boost::vecS, boost::undirectedS, RDKit::Atom*, RDKit::Bond*>, long unsigned int, boost::detail::out_edge_iter<__gnu_cxx::__normal_iterator<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*>*, std::vector<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*>, std::allocator<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*> > > >, long unsigned int, boost::detail::edge_desc_impl<boost::undirected_tag, long unsigned int>, long int>, long int>]':
/home/postgres/pgsource/external/rdkit/Code/GraphMol/CIPLabeler/CIPMol.h:55:47:   required from 'RDKit::CIPLabeler::CIPMolSpan<T, U>::CIPMolIter RDKit::CIPLabeler::CIPMolSpan<T, U>::begin() [with T = RDKit::Atom*; U = boost::adjacency_iterator<boost::adjacency_list<boost::vecS, boost::vecS, boost::undirectedS, RDKit::Atom*, RDKit::Bond*>, long unsigned int, boost::detail::out_edge_iter<__gnu_cxx::__normal_iterator<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*>*, std::vector<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*>, std::allocator<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*> > > >, long unsigned int, boost::detail::edge_desc_impl<boost::undirected_tag, long unsigned int>, long int>, long int>]'
/home/postgres/pgsource/external/rdkit/Code/GraphMol/CIPLabeler/Mancude.cpp:108:49:   required from here
/home/postgres/pgsource/external/rdkit/Code/GraphMol/CIPLabeler/CIPMol.h:29:69: error: invalid initialization of non-const reference of type 'RDKit::ROMol&' from an rvalue of type '<brace-enclosed initializer list>'
/home/postgres/pgsource/external/rdkit/Code/GraphMol/SLNParse/SLNParse.cpp: In function 'std::string RDKit::SLNParse::replaceSLNMacroAtoms(std::string, int)':
/home/postgres/pgsource/external/rdkit/Code/GraphMol/SLNParse/SLNParse.cpp:126:77: error: no matching function for call to 'regex_replace(std::string&, const regex&, const char*&, std::bitset<11ul>)'
       std::regex_constants::match_default | std::regex_constants::format_sed);
                                                                             ^
/home/postgres/pgsource/external/rdkit/Code/GraphMol/SLNParse/SLNParse.cpp:126:77: note: candidates are:
In file included from /usr/include/c++/4.8.2/regex:62:0,
                 from /home/postgres/pgsource/external/rdkit/Code/GraphMol/SLNParse/SLNParse.cpp:42:
/usr/include/c++/4.8.2/bits/regex.h:2162:5: note: template<class _Out_iter, class _Bi_iter, class _Rx_traits, class _Ch_type> _Out_iter std::regex_replace(_Out_iter, _Bi_iter, _Bi_iter, const std::basic_regex<_Ch_type, _Rx_traits>&, const std::basic_string<_Ch_type>&, std::regex_constants::match_flag_type)
     regex_replace(_Out_iter __out, _Bi_iter __first, _Bi_iter __last,
     ^
/usr/include/c++/4.8.2/bits/regex.h:2162:5: note:   template argument deduction/substitution failed:
/home/postgres/pgsource/external/rdkit/Code/GraphMol/SLNParse/SLNParse.cpp:126:77: note:   deduced conflicting types for parameter '_Bi_ite' ('std::basic_regex<char>' and 'const char*')
       std::regex_constants::match_default | std::regex_constants::format_sed);
                                                                             ^
In file included from /usr/include/c++/4.8.2/regex:62:0,
                 from /home/postgres/pgsource/external/rdkit/Code/GraphMol/SLNParse/SLNParse.cpp:42:
/usr/include/c++/4.8.2/bits/regex.h:2182:5: note: template<class _Rx_traits, class _Ch_type> std::basic_string<_Ch_type> std::regex_replace(const std::basic_string<_Ch_type>&, const std::basic_regex<_Ch_type, _Rx_traits>&, const std::basic_string<_Ch_type>&, std::regex_constants::match_flag_type)
     regex_replace(const basic_string<_Ch_type>& __s,
     ^
/usr/include/c++/4.8.2/bits/regex.h:2182:5: note:   template argument deduction/substitution failed:
/home/postgres/pgsource/external/rdkit/Code/GraphMol/SLNParse/SLNParse.cpp:126:77: note:   mismatched types 'const std::basic_string<_Ch_type>' and 'const char*'
       std::regex_constants::match_default | std::regex_constants::format_sed);
                                                                             ^
/home/postgres/pgsource/external/rdkit/Code/GraphMol/SLNParse/SLNParse.cpp:142:68: error: no matching function for call to 'regex_replace(std::string&, std::regex, const char*, std::bitset<11ul>)'
                                    std::regex_constants::format_sed);
                                                                    ^
/home/postgres/pgsource/external/rdkit/Code/GraphMol/SLNParse/SLNParse.cpp:142:68: note: candidates are:
In file included from /usr/include/c++/4.8.2/regex:62:0,
                 from /home/postgres/pgsource/external/rdkit/Code/GraphMol/SLNParse/SLNParse.cpp:42:
/usr/include/c++/4.8.2/bits/regex.h:2162:5: note: template<class _Out_iter, class _Bi_iter, class _Rx_traits, class _Ch_type> _Out_iter std::regex_replace(_Out_iter, _Bi_iter, _Bi_iter, const std::basic_regex<_Ch_type, _Rx_traits>&, const std::basic_string<_Ch_type>&, std::regex_constants::match_flag_type)
     regex_replace(_Out_iter __out, _Bi_iter __first, _Bi_iter __last,
     ^
/usr/include/c++/4.8.2/bits/regex.h:2162:5: note:   template argument deduction/substitution failed:
/home/postgres/pgsource/external/rdkit/Code/GraphMol/SLNParse/SLNParse.cpp:142:68: note:   deduced conflicting types for parameter '_Bi_ite' ('std::basic_regex<char>' and 'const char*')
                                    std::regex_constants::format_sed);
                                                                    ^
In file included from /usr/include/c++/4.8.2/regex:62:0,
                 from /home/postgres/pgsource/external/rdkit/Code/GraphMol/SLNParse/SLNParse.cpp:42:
/usr/include/c++/4.8.2/bits/regex.h:2182:5: note: template<class _Rx_traits, class _Ch_type> std::basic_string<_Ch_type> std::regex_replace(const std::basic_string<_Ch_type>&, const std::basic_regex<_Ch_type, _Rx_traits>&, const std::basic_string<_Ch_type>&, std::regex_constants::match_flag_type)
     regex_replace(const basic_string<_Ch_type>& __s,
     ^
/usr/include/c++/4.8.2/bits/regex.h:2182:5: note:   template argument deduction/substitution failed:
/home/postgres/pgsource/external/rdkit/Code/GraphMol/SLNParse/SLNParse.cpp:142:68: note:   mismatched types 'const std::basic_string<_Ch_type>' and 'const char*'
                                    std::regex_constants::format_sed);

**Comment by ptosco** [comment]
g++ 4.8 is very old, and only has experimental support for C++11. You will need to use a more recent compiler version.

**Reply by Joezczhang** [reply]
Hello:

I have many machines based on g++ 4.8, for me use a more recent compiler version in only a part of machines is not the first choice .

Is there any possibility to use rdkit with g++ 4.8.

Here is introduction from https://github.com/rdkit/rdkit/blob/master/Docs/Book/Install.md
<img width="1059" alt="image" src="https://user-images.githubusercontent.com/89260661/131679381-b6b3ab08-5673-4465-a2c4-e159472c7ac2.png">

I want rdkit just like it(without SLN).

**Comment by ptosco** [comment]
@Joezczhang Are you using CentOS 7 (or some other RH7-based distro)? If so, you can install https://www.softwarecollections.org/en/scls/rhscl/devtoolset-8/ only on the machine where you build RDKit. This will give you a more recent compiler version which is entirely compatible with CentOS 7 `libstdc++`. The built libraries will still work on vanilla CentOS 7 machines where the devtoolset is not installed.
I haven't attempted building RDKit with `g++ 4.8.5`. It looks like the problem affects a single file (apart from SLN). I'll have a look and see if there is a simple fix.

**Comment by Joezczhang** [comment]
Hi:
Are you using CentOS 7 (or some other RH7-based distro)? -> Yes.

Thanks a lot for your information.

**Comment by greglandrum** [comment]
I do suggest using the more up-to-date compilers provided by the devtoolset, but if you disable SLN building and use boost regex, the two cmake flags are `-DRDK_USE_BOOST_REGEX=ON -DRDK_BUILD_SLN_SUPPORT=OFF` then I think you should be able to get a successful build. Some tests require `std::regex` to work (which it most definitely does *not* with g++ 4.8), so you will get some test failures related to that.

**Comment by Joezczhang** [comment]
Hello greglandrum:

I did your suggestion but there are still build errors just like below.No matter what I did, those errors will occur with g++ 4.85

/home/postgres/rdkit/source/v12.4/contrib/rdkit/Code/GraphMol/CIPLabeler/CIPMol.cpp: In constructor 'RDKit::CIPLabeler::CIPMol::CIPMol(RDKit::ROMol&)':
/home/postgres/rdkit/source/v12.4/contrib/rdkit/Code/GraphMol/CIPLabeler/CIPMol.cpp:19:39: error: invalid initialization of non-const reference of type 'RDKit::ROMol&' from an rvalue of type '<brace-enclosed initializer list>'
 CIPMol::CIPMol(ROMol &mol) : d_mol{mol} {}
                                       ^
/home/postgres/rdkit/source/v12.4/contrib/rdkit/Code/GraphMol/CIPLabeler/CIPMol.cpp: In constructor 'RDKit::CIPLabeler::CIPMol::CIPMol(RDKit::ROMol&)':
/home/postgres/rdkit/source/v12.4/contrib/rdkit/Code/GraphMol/CIPLabeler/CIPMol.cpp:19:39: error: invalid initialization of non-const reference of type 'RDKit::ROMol&' from an rvalue of type '<brace-enclosed initializer list>'
 CIPMol::CIPMol(ROMol &mol) : d_mol{mol} {}
                                       ^
/home/postgres/rdkit/source/v12.4/contrib/rdkit/Code/GraphMol/MolInterchange/Parser.cpp:30:32: warning: unknown option after '#pragma GCC diagnostic' kind [-Wpragmas]
 #pragma GCC diagnostic ignored "-Wclass-memaccess"
                                ^
In file included from /home/postgres/rdkit/source/v12.4/contrib/rdkit/Code/GraphMol/CIPLabeler/CIPMol.cpp:14:0:
/home/postgres/rdkit/source/v12.4/contrib/rdkit/Code/GraphMol/CIPLabeler/CIPMol.h: In instantiation of 'RDKit::CIPLabeler::CIPMolSpan<T, U>::CIPMolSpan(RDKit::ROMol&, std::pair<OutEdgeIterator, OutEdgeIterator>&&) [with T = RDKit::Bond*; U = boost::detail::out_edge_iter<__gnu_cxx::__normal_iterator<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*>*, std::vector<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*>, std::allocator<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*> > > >, long unsigned int, boost::detail::edge_desc_impl<boost::undirected_tag, long unsigned int>, long int>]':
/home/postgres/rdkit/source/v12.4/contrib/rdkit/Code/GraphMol/CIPLabeler/CIPMol.cpp:42:42:   required from here
/home/postgres/rdkit/source/v12.4/contrib/rdkit/Code/GraphMol/CIPLabeler/CIPMol.h:53:37: error: invalid initialization of non-const reference of type 'RDKit::ROMol&' from an rvalue of type '<brace-enclosed initializer list>'
         d_iend{std::move(itr.second)} {}
                                     ^
/home/postgres/rdkit/source/v12.4/contrib/rdkit/Code/GraphMol/CIPLabeler/CIPMol.h: In instantiation of 'RDKit::CIPLabeler::CIPMolSpan<T, U>::CIPMolSpan(RDKit::ROMol&, std::pair<OutEdgeIterator, OutEdgeIterator>&&) [with T = RDKit::Atom*; U = boost::adjacency_iterator<boost::adjacency_list<boost::vecS, boost::vecS, boost::undirectedS, RDKit::Atom*, RDKit::Bond*>, long unsigned int, boost::detail::out_edge_iter<__gnu_cxx::__normal_iterator<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*>*, std::vector<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*>, std::allocator<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*> > > >, long unsigned int, boost::detail::edge_desc_impl<boost::undirected_tag, long unsigned int>, long int>, long int>]':
/home/postgres/rdkit/source/v12.4/contrib/rdkit/Code/GraphMol/CIPLabeler/CIPMol.cpp:47:46:   required from here
/home/postgres/rdkit/source/v12.4/contrib/rdkit/Code/GraphMol/CIPLabeler/CIPMol.h:53:37: error: invalid initialization of non-const reference of type 'RDKit::ROMol&' from an rvalue of type '<brace-enclosed initializer list>'
In file included from /home/postgres/rdkit/source/v12.4/contrib/rdkit/Code/GraphMol/CIPLabeler/CIPMol.cpp:14:0:
/home/postgres/rdkit/source/v12.4/contrib/rdkit/Code/GraphMol/CIPLabeler/CIPMol.h: In instantiation of 'RDKit::CIPLabeler::CIPMolSpan<T, U>::CIPMolSpan(RDKit::ROMol&, std::pair<OutEdgeIterator, OutEdgeIterator>&&) [with T = RDKit::Bond*; U = boost::detail::out_edge_iter<__gnu_cxx::__normal_iterator<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*>*, std::vector<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*>, std::allocator<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*> > > >, long unsigned int, boost::detail::edge_desc_impl<boost::undirected_tag, long unsigned int>, long int>]':
/home/postgres/rdkit/source/v12.4/contrib/rdkit/Code/GraphMol/CIPLabeler/CIPMol.cpp:42:42:   required from here
/home/postgres/rdkit/source/v12.4/contrib/rdkit/Code/GraphMol/CIPLabeler/CIPMol.h:53:37: error: invalid initialization of non-const reference of type 'RDKit::ROMol&' from an rvalue of type '<brace-enclosed initializer list>'
         d_iend{std::move(itr.second)} {}
                                     ^
/home/postgres/rdkit/source/v12.4/contrib/rdkit/Code/GraphMol/CIPLabeler/CIPMol.h: In instantiation of 'RDKit::CIPLabeler::CIPMolSpan<T, U>::CIPMolSpan(RDKit::ROMol&, std::pair<OutEdgeIterator, OutEdgeIterator>&&) [with T = RDKit::Atom*; U = boost::adjacency_iterator<boost::adjacency_list<boost::vecS, boost::vecS, boost::undirectedS, RDKit::Atom*, RDKit::Bond*>, long unsigned int, boost::detail::out_edge_iter<__gnu_cxx::__normal_iterator<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*>*, std::vector<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*>, std::allocator<boost::detail::stored_edge_iter<long unsigned int, std::_List_iterator<boost::list_edge<long unsigned int, RDKit::Bond*> >, RDKit::Bond*> > > >, long unsigned int, boost::detail::edge_desc_impl<boost::undirected_tag, long unsigned int>, long int>, long int>]':
/home/postgres/rdkit/source/v12.4/contrib/rdkit/Code/GraphMol/CIPLabeler/CIPMol.cpp:47:46:   required from here
/home/postgres/rdkit/source/v12.4/contrib/rdkit/Code/GraphMol/CIPLabeler/CIPMol.h:53:37: error: invalid initialization of non-const reference of type 'RDKit::ROMol&' from an rvalue of type '<brace-enclosed initializer list>'
